packages/ra-core/src/actions/dataActions/crudGetAll.ts

```diff
@@ -0,0 +1,88 @@
+import { GET_LIST } from '../../dataFetchActions';
+import { FETCH_END, FETCH_ERROR } from '../fetchActions';
+import { NotificationSideEffect } from '../../sideEffect/notification';
+
+interface RequestPayload {
+    pagination: { page: number; perPage: number };
+    sort: { field: string; order: string };
+    filter: object;
+}
+
+export const CRUD_GET_ALL = 'RA/CRUD_GET_ALL';
+interface CrudGetAllAction {
+    readonly type: typeof CRUD_GET_ALL;
+    readonly payload: RequestPayload;
+    readonly meta: {
+        resource: string;
+        fetch: typeof GET_LIST;
+        onFailure: {
+            notification: NotificationSideEffect;
+        };
+        onSuccess: {
+            callback: any;
+        };
+    };
+}
+
+export const crudGetAll = (
+    resource: string,
+    sort: { field: string; order: string },
+    filter: object,
+    maxResults: number,
+    callback?: () => any
```

Comment:
As we know what will be the callback params, we should add them too:
`{ payload, requestPayload }`